PowerEditor/src/Notepad_plus_Window.cpp

```diff
@@ -252,6 +252,7 @@ void Notepad_plus_Window::init(HINSTANCE hInst, HWND parent, const TCHAR *cmdLin
 		// the Project Panels are not loaded by Notepad_plus::doOpen only, but also by the Plugin Manager restoring the state
 		// of the last session from config.xml.
 		::SendMessage(_hSelf, WM_COMMAND, IDM_VIEW_PROJECT_PANEL_1, 0);
+		::SendMessage(_hSelf, WM_COMMAND, IDM_VIEW_SWITCHTO_OTHER_VIEW, 0);

```

Comment:
What are you trying to do here?

Reply:
I switch the focus to the edit view after switching to the correct Project panel, and this only while starting Npp with a workspace file name in the command line.

This is to achieve a similar behavior to other IDE's, easiest example here is MS VC++. The solution file .sln is the equivalent to the workspace file, and if it is opened with MS VC++, the input focus is not in the "Solution explorer", but in the edit window.

Of course, the message I used to achieve this was chosen more to make the code line look similar to the previous one. I looked for what was available in the `Notepad_plus::command` function and found nothing better.

Since I get deeper into it day by day, I can now offer:

		::SendMessage(_hSelf, WM_ACTIVATE, WA_ACTIVE, 0);

And, we could open the big packet and compare the behavior of 
- _Folder as Workspace_
- _Project Panel_
- _Function List_
- _Document Switcher_

What we see is, that the _Function List_ switches back to the edit window in the `Notepad_plus::launchFunctionList()` function, the other three do not. So what happens is, that the input focus tends to be in one of those three side windows instead of in the edit window after startup. Now we could insert a `_pEditView->getFocus()` call at the end of
- `Notepad_plus::launchFileBrowser()`
- `Notepad_plus::launchProjectPanel()`
- `Notepad_plus::launchFileSwitcherPanel()`
(It is already there in `Notepad_plus::launchFunctionList()`)

Then we wouldn't need a separate focus change message while startup. But, it would change the behavior of the three panels when they are started from the menu. Today, after starting them from the menu, they have the input focus. If we make them behave like the _Function List_, then the input focus will be in the edit window after starting them from the menu.

So I asked the MS VC++ IDE here, closed the "Solution Explorer" and opened it again with "View->Solution Explorer", result: The "Solution Explorer" has the input focus.

Which means, the current behavior may not be unexpected by many users. Based on this, I would give the input focus to the side window when it is launched from the menu. But I would also expect, that the edit window has the input focus after startup. The easiest way to solve all of it is to move the `WM_ACTIVATE` message one bracket down. Like this:

	if (cmdLineParams->_openFoldersAsWorkspace)
	{
		_notepad_plus_plus_core.launchFileBrowser(fns, true);
	}
	else if (_notepad_plus_plus_core._isWorkspaceFileLoadedFromCommandLine)
	{
		::SendMessage(_hSelf, WM_COMMAND, IDM_VIEW_PROJECT_PANEL_1, 0);
	}
	::SendMessage(_hSelf, WM_ACTIVATE, WA_ACTIVE, 0);

It fixes the startup input focus for all three panels without changing there current behavior while starting them from the menu. Additionally, you could think about removing the `_pEditView->getFocus()` call from `Notepad_plus::launchFunctionList()`, to make the _Function List_ behave like the other panels.

Alternatively, we could put the `_pEditView->getFocus()` in the `launch...` functions of the three other panels and accept the change of behavior while starting them from the menu. Would be fine with me too.

Many choices. What would you like?

Reply:
`IDM_VIEW_SWITCHTO_OTHER_VIEW` makes the focus change from a Scintilla Edit View to another one, if you have 2 edit views opened side by side. So I don't think it's what you want.

Reply:
You are right, that's not what I want to do. But `IDM_VIEW_SWITCHTO_OTHER_VIEW` works, because it switches the input focus to the current edit view, as long as the input focus is not at the current edit view. And in this situation, the input focus is at the Project panel.

But as I wrote, I would use `WM_ACTIVATE` now instead, and I would make it work for _Folder As Workspace_ and _Doc Switcher_ too, by shifting it one line down.

Reply:
> But IDM_VIEW_SWITCHTO_OTHER_VIEW works,

Just as having sex with my sister - it works doesn't mean it's right :)

> But as I wrote, I would use WM_ACTIVATE now instead,

Yes, please do it.

Reply:
Do you approve the described move of the `WM_ACTIVATE` message one line down too? This would make the following change to the original PR: During startup, the input focus would be set to the edit window not only in the case when a Workspace file is specified in the command line, but also if any of the following side windows is loaded during startup:
- _Folder as Workspace_
- _Project Panel_
- _Doc Switcher_

The current state is, that if any of this side windows is loaded automatically during startup, the input focus is in the side window instead of the edit window. This would change for good if we move the `WM_ACTIVATE` line as I described in my long comment.

Reply:
@Uhf7 
> This would make the following change to the original PR: During startup, the input focus would be set to the edit window not only in the case when a Workspace file is specified in the command line, but also if any of the following side windows is loaded during startup:

Yes, the focus should be always on the edit view after startup.